hw5/code/hw6_script.py

```diff
@@ -0,0 +1,120 @@
+###############################################################################################
+# this script is a copy of a notebook with the 6th homework for pandas and matplotlib/seaborn #
+###############################################################################################
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+## the first part: pandas
+
+# let's read *.ggf
+def read_gff(path):
+    return pd.read_csv(path, sep="\t", comment="#",
+                       names=["chromosome", "source", "type", "start", "end",
+                              "score", "strand", "phase", "attributes"])
+
+
+gff = read_gff('rrna_annotation.gff')
+
+
+# let's read *.bed
+
+def read_bed6(path):
+    names = ['chromosome', 'start', 'end', 'name', 'score', 'strand']
+    return pd.read_csv(path, sep='\t', names=names)
+
+
+bed6 = read_bed6('alignment.bed')
+
+# reduce amount of information in the 'attributes' column
+gff['attributes'] = gff['attributes'].apply(lambda x: x.split('Name=')[1].split('_')[0])
+
+# group by types of rRNA and chromosomes
+gff2 = gff.iloc[:, [0, -1]]
+gff2.columns = ['Sequence', 'RNA type']
+gff2 = gff2.groupby(by=['Sequence', 'RNA type']).size()
+
+# just a more appealing way to represent the data
+gff_cut = gff2.reset_index()
+gff_cut = gff_cut.pivot(index='Sequence', columns='RNA type', values=0)
+gff_cut = gff_cut.replace(np.nan, 0)
+gff_cut.astype('int32').transpose()
+
+# a barplot
+gff2.unstack().plot.bar(figsize=(12, 5))
+
+# a bedtools analog
+new = pd.merge(gff, bed6, on="chromosome", how="outer")
+new[(new['start_x'] >= new['start_y']) & (new['end_x'] <= new['end_y'])]
+
+
+
+## the second part: matplotlib/seaborn
+de = pd.read_csv('diffexpr_data.tsv.gz', sep='\t')
```

Comment:
Нейминг! Не нужно жалеть символы и смысл на имена переменных, это относится ко всем заданиям
```suggestion
diffexpr_data = pd.read_csv('diffexpr_data.tsv.gz', sep='\t')
```